linkatos/message.py

```diff
@@ -21,3 +25,58 @@ def has_a_yes(message):
     Returns True if it matches the yes regex
     """
     return yes_re.search(message) is not None
+
+
+def has_a_no(message):
+    """
+    Returns True if it matches the no regex
+    """
+    return no_re.search(message) is not None
+
+
+def parse(slack_rtm_output):
+    """
+        The Slack Real Time Messaging API is an events firehose.
+        this parsing function returns None unless:
+        1. someone posts a url starting with httpS?//
+           in this case it returns the url and a out_type = 'url'
+        2. a yes or a no after a url
+           in this case it returns True for a yes, False for a no
+        The function also extracts the name of the channel
+    """
+    # default outcome
+
+    output_list = slack_rtm_output
+    print(output_list)  # print the list of outputs to get them on screen
+
+    if (output_list is None) or (len(output_list) == 0):
+        return (None, None, None)
+
+    for output in output_list:
+        if output is None or 'text' not in output or output['user'] == BOT_ID:
+            print("output none or text not there or user bot")
+            return (None, None, None)
+
+        text = output['text']
```

Comment:
I have the feeling this function tries to do too much.  Wouldn't be better to isolate the logic for each case? at least group it and return for each branch instead of composing the final message here and there.